### PR TITLE
Opt Out Advertising: check consent when in test

### PR DESCRIPTION
**Pull request by chrislomaxjones**
Co-authored-by: Simon Byford <simon.byford@guardian.co.uk>

## What does this change?

This PR changes the behaviour in the consentless ads 0% test to take into account a browser's consent state to decide which third party advertising tag to launch (Googletag or Opt Out). Prior to this PR, we launched the Opt Out tag when you opt into the 0% AB test, regardless of your consent state.

We use consent state to determine which tag to launch. The rules are as follows:
- Always launch Googletag when in CCPA and Australia (as there's no way to opt out of a specific vendor).
- When in TCF, we check whether the browser has consented to the `googletag` vendor. If they have, we launch Googletag and if not we launch Opt Out.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

Prior to this PR, we launch the Opt Out tag when you opt into the 0% AB test, regardless of your consent state. Now, when in the variant, this will take into account your consent state and decided which tag to launch based on that. This will mimic the behaviour of the commercial code as we roll out the test to a larger audience.

### Tested 

**TODO**

- [ ] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by simonbyford — APPROVED** [review]
🥳

**Review by domlander — APPROVED** [review]

**Review by Jakeii — APPROVED** [review]
nice

File: static/src/javascripts/bootstraps/standalone.commercial.ts [hunk 1]
```diff
@@ -205,15 +209,33 @@ const bootCommercial = async (): Promise<void> => {
 	}
 };
 
+/**
+ * Choose whether to launch Googletag or Opt Out tag (ootag) based on consent state
+ */
+const chooseAdvertisingTag = async () => {
```
Comment (arelra) [inline static/src/javascripts/bootstraps/standalone.commercial.ts:215]: This is quite a big change to the loading process of commercial. Are we able to analyse any potential performance impact commercial metrics (particularly with Tableau not being available)?

**Review by arelra — COMMENTED** [review]
🚀

File: static/src/javascripts/bootstraps/standalone.commercial.ts (hunk 1 above)
Comment (chrislomaxjones) [reply static/src/javascripts/bootstraps/standalone.commercial.ts:215]: As discussed, we'll look to address this in subsequent work, but this isn't a blocker whilst we're still running at zero percent.

**Pull request merged**